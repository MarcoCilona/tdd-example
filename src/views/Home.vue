<template>
  <div>
    <NameForm @submit='handleSubmit'/>
    <NameView :name='name' />
  </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

// Components
import NameForm from '@/components/NameForm.vue';
import NameView from '@/components/NameView.vue';

@Component(
  {
    components: {
      NameForm,
      NameView,
    },
  },
)
export default class Home extends Vue {
  name = '';

  handleSubmit($event: string) {
    this.name = $event;
  }
}
</script>

<style>
</style>
