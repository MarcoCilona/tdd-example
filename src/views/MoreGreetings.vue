<template>
  <div>
    <h1 data-set-id="titleTest">{{ name }}</h1>
    <p data-set-id="paragraphGreetingsTest">Watch out, {{name}} is coming</p>
  </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

@Component
export default class MoreGreetings extends Vue {
  name = '';

  mounted() {
    this.name = this.$route.params.name;
  }
}
</script>
