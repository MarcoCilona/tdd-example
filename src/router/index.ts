import Vue from 'vue';
import VueRouter, { RouteConfig } from 'vue-router';

import Home from '@/views/Home.vue';
import MoreGreetings from '@/views/MoreGreetings.vue';

Vue.use(VueRouter);

export const routes: Array<RouteConfig> = [
  {
    path: '/',
    component: Home,
  },
  {
    path: '/more-greetings/:name',
    name: 'MoreGreetings',
    component: MoreGreetings,
  },
];

const router = new VueRouter({
  routes,
});

export default router;
