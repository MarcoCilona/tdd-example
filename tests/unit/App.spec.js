import { createLocalVue, mount } from '@vue/test-utils';
import App from '@/App.vue';

import VueRouter from 'vue-router';
import { routes } from '@/router/index';

//  Components
import Home from '@/views/Home.vue';
import MoreGreetings from '@/views/MoreGreetings.vue';

const localVue = createLocalVue();
localVue.use(VueRouter);

/**
 * Test needed:
 * 1. Componente App exists and its the app container;
 * 2. First access App is pointing to '/' route
 * 3. App contains Home component
 * 4. When route changes to '/more-greetings' App contains MoreGreetings component
 */
describe('App tests', () => {
  let wrapper;
  let router;

  beforeEach(() => {
    router = new VueRouter({ routes });
    wrapper = mount(App, {
      localVue,
      router,
      data: () => ({
        name: '',
      }),
    });
  });

  it('App component exists and is rendered right', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('App components its the main application wrapper', () => {
    expect(wrapper.find('#app').exists()).toBe(true);
  });

  it('First access App is pointing to "/" route', () => {
    expect(wrapper.vm.$route.path).toEqual('/');
  });

  it('First access App contains Home component', () => {
    expect(wrapper.findComponent(Home).exists()).toBe(true);
  });

  it('When routes is "/more-greetings" the App contains MoreGreetings component', async () => {
    router.push('/more-greetings/name');
    await wrapper.vm.$nextTick();
    expect(wrapper.findComponent(MoreGreetings).exists()).toBe(true);
  });
});
