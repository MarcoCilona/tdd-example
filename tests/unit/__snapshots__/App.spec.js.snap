// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests App component exists and is rendered right 1`] = `
<div id="app">
  <div>
    <div><input data-test-id="inputTest"> <button data-test-id="buttonTest">Click me!</button></div>
    <div>
      <p data-test-id="paragraphTest">Hello </p> <button data-test-id="greetingsButton">More greeting please!</button>
    </div>
  </div>
</div>
`;
